.github/workflows/build-test-linux.yml

```diff
@@ -30,7 +30,7 @@ jobs:
         go-version: ~1.15.15
 
     - name: Check out code
-      uses: actions/checkout@v2
```

Comment:
Its fine to separate in a separate PR since I only want to know if the `cache@v3` would [solve the cache our artifacts when building in Linux](https://github.com/aws/amazon-cloudwatch-agent/pull/480/checks)